### PR TITLE
Fix auto_ptr deprecations

### PR DESCRIPTION
**Pull request by OrelSokolov**
We have a lot of deprecations when trying to build it.

### CODE REVIEW

File: cmake/SetDefaultGccFlags.cmake [hunk 1]
```diff
@@ -7,7 +7,19 @@ MACRO(ST_SET_DEFAULT_GCC_FLAGS)
 		SET(gc_sections_cflags_ "")
 		SET(gc_sections_ldflags_ "")
 		SET(no_inline_dllexport_cflags_ "")
-		
+		SET(cpp11_flags_ "")
+
+		CHECK_CXX_ACCEPTS_FLAG(
+				"-ansi"
```
Comment (eyalroz) [inline cmake/SetDefaultGccFlags.cmake]: `-ansi` is for C++98. Are you sure you need to check for that?

File: cmake/SetDefaultGccFlags.cmake [hunk 2]
```diff
@@ -7,7 +7,19 @@ MACRO(ST_SET_DEFAULT_GCC_FLAGS)
...
+				cpp11_support_
+		)
+		MESSAGE( STATUS "Current CPP11 flag: " ${cpp11_support_} )
+		IF(cpp11_support_)
+			SET(cpp11_flags_ "-std=c++11")
```
Comment (eyalroz) [inline cmake/SetDefaultGccFlags.cmake:17]: CMake has builtin mechanisms for dealing with language standard versions; see [here](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html).

File: cmake/SetDefaultGccFlags.cmake (hunk 1 above)
Comment (OrelSokolov) [reply cmake/SetDefaultGccFlags.cmake]: Oh, sorry, here is a bit of garbage. I am almost noob in C++ projects. There were just first attempts to add check for failing examples.

File: cmake/SetDefaultGccFlags.cmake (hunk 2 above)
Comment (OrelSokolov) [reply cmake/SetDefaultGccFlags.cmake:17]: Oh, cool.

File: cmake/SetDefaultGccFlags.cmake (hunk 1 above)
Comment (OrelSokolov) [reply cmake/SetDefaultGccFlags.cmake]: Actually I could not understand why my check does not work. It was temporary check for some well known flag. Thanks.

**Comment by eyalroz** [comment]
Yeah, auto_ptr deprecation warnings are annoying, because you get a zillion of them...

However, note that people often had to structure their classes to handle aspects of `auto_ptr`'s behavior, e.g. use custom ctor's, dtor's, assignment operators etc - which are obviated by the switch to `std::unique_ptr`. In case you haven't done so, I'd try to check whether there are cases of this having happened, which can now be simplified somewhat.

**Comment by OrelSokolov** [comment]
@eyalroz what is `ctor's and dtor's` ? Constructors and destructors?

**Comment by eyalroz** [comment]
> @eyalroz what is `ctor's and dtor's` ? Constructors and destructors?

Yes, that's right.

**Pull request merged**